tests/ChildRuleTest.php

```diff
@@ -10,97 +10,229 @@
+  }
+
   protected function render(): XHPRoot {
     return <div />;
   }
 }
 
 class :test:two-children extends :x:element {
+  use XHPChildDeclarationConsistencyValidation;
   children (:div, :div);
+  protected static function getChildrenDeclaration(): XHPChild\Constraint {
+    return XHPChild\sequence(XHPChild\ofType<:div>(), XHPChild\ofType<:div>());
+  }
+
+  protected function render(): XHPRoot {
+    return <div />;
+  }
+}
+
+class :test:three-children extends :x:element {
+  use XHPChildDeclarationConsistencyValidation;
+  children (:div, :div, :div);
+  protected static function getChildrenDeclaration(): XHPChild\Constraint {
+    return XHPChild\sequence(
+      XHPChild\ofType<:div>(),
+      XHPChild\ofType<:div>(),
+      XHPChild\ofType<:div>(),
+    );
+  }
+
   protected function render(): XHPRoot {
     return <div />;
   }
 }
 
+
 class :test:either-of-two-children extends :x:element {
+  use XHPChildDeclarationConsistencyValidation;
   children (:div | :code);
+  protected static function getChildrenDeclaration(): XHPChild\Constraint {
+    return XHPChild\anyOf(XHPChild\ofType<:div>(), XHPChild\ofType<:code>());
+  }
+
+  protected function render(): XHPRoot {
+    return <div />;
+  }
+}
+
+class :test:any-of-three-children extends :x:element {
+  use XHPChildDeclarationConsistencyValidation;
+  children (:div | :code | :p);
+  protected static function getChildrenDeclaration(): XHPChild\Constraint {
+    return XHPChild\anyOf(
+      XHPChild\ofType<:div>(),
+      XHPChild\ofType<:code>(),
+      XHPChild\ofType<:p>(),
+    );
+  }
+
   protected function render(): XHPRoot {
     return <div />;
   }
 }
 
+
 class :test:nested-rule extends :x:element {
+  use XHPChildDeclarationConsistencyValidation;
   children (:div | (:code+));
+  protected static function getChildrenDeclaration(): XHPChild\Constraint {
+    return XHPChild\anyOf(
+      XHPChild\ofType<:div>(),
+      XHPChild\atLeastOneOf(XHPChild\ofType<:code>()),
+    );
+  }
+
   protected function render(): XHPRoot {
     return <div />;
   }
 }
 
 class :test:pcdata-child extends :x:element {
+  use XHPChildDeclarationConsistencyValidation;
   children (pcdata);
+  protected static function getChildrenDeclaration(): XHPChild\Constraint {
+    return XHPChild\pcdata();
+  }
 
   protected function render(): XHPRoot {
     return <div>{$this->getChildren()}</div>;
   }
 }
 
 class :test:category-child extends :x:element {
+  use XHPChildDeclarationConsistencyValidation;
   children (%flow);
+  protected static function getChildrenDeclaration(): XHPChild\Constraint {
+    return XHPChild\category('%flow');
+  }
 
   protected function render(): XHPRoot {
     return <div />;
   }
 }
 
 class :test:has-comma-category extends :x:element {
+  use XHPChildDeclarationConsistencyValidation;
   category %foo:bar;
+  protected static function getChildrenDeclaration(): XHPChild\Constraint {
+    return XHPChild\category('%foo:bar');
+  }
```

Comment:
Looks like you added this when you didn't need to, since it was a category definition instead of a children definition.